nixos/modules/services/logging/logrotate.nix

```diff
@@ -130,26 +171,46 @@ in
               path = "/var/log/httpd/*.log";
               user = config.services.httpd.user;
               group = config.services.httpd.group;
-              keep = 7;
+              rotate = 7;
             };
 
             myapp = {
               path = "/var/log/myapp/*.log";
               user = "myuser";
               group = "mygroup";
               frequency = "weekly";
-              keep = 5;
+              rotate = 5;
               priority = 1;
             };
           }
         '';
       };
 
+      checkConfig = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether the config should be checked at build time.
+          Some options might not be checkable at build time for some reason
+          (sandbox or wrong user)
+          Disable check if it can't be fixed.
+        '';
+      };
+
       extraConfig = mkOption {
-        default = "";
-        type = types.lines;
+        type = with types; nullOr lines;
+        default = null;
+        description = ''
+          Extra lines prepended to logrotate configuration file.
+          Deprecated, use settings instead.
```

Comment:
For a prettier rendering, you may use: `<xref linkend="opt-services.logrotate.settings"/>`